File: CoffeeMaker/Python/make_coffee.py
from weak_coffee import *
from wet_beans_coffee import *
from dry_beans_coffee import *
from coffee_maker import *


class MakingCoffee:
    def make_a_coffee(self):
        make_weak_coffee = CoffeeMaker(WeaKCoffee())
        make_wet_coffee = CoffeeMaker(DryBeansCoffee())
        make_dry_coffee = CoffeeMaker(WetBeansCoffee())

        print(make_weak_coffee.making_coffee())
        print(make_wet_coffee.making_coffee())
        print(make_dry_coffee.making_coffee())


if __name__ == '__main__':
    make_coffee = MakingCoffee()
    make_coffee.make_a_coffee()

File: Mocking/CoffeeMaker/CSHARP/CoffeeMaker/makeCoffee.cs
using System;
namespace CoffeeMaker
{
    public class MakeCoffee
    {
        public MakeCoffee()
        {

        }
        public void MakeAllCoffees()
        {
            ICoffeeSelect weakCoffee = new WeakCoffee();
            ICoffeeSelect wetCoffee = new DryBeansCoffee();
            ICoffeeSelect dryCoffee = new WetBeansCoffee();

            ICoffeeMaker weakCoffeeMaker = new CoffeeMaker(weakCoffee);
            ICoffeeMaker dryCoffeeMaker = new CoffeeMaker(dryCoffee);
            ICoffeeMaker wetCoffeeMaker = new CoffeeMaker(wetCoffee);

            Console.WriteLine(weakCoffeeMaker.MakingCoffee());
            Console.WriteLine(dryCoffeeMaker.MakingCoffee());
            Console.WriteLine(wetCoffeeMaker.MakingCoffee());
        }
    }
}
